core/src/java/test/org/jaxen/test/BaseXPathTest.java

```diff
@@ -1090,8 +1090,8 @@ public void testAttributeNodesOnParentComeBeforeNamespaceNodesInChildInDocumentO
         XPath xpath = new DOMXPath("/*/*/namespace::node() | //attribute::* ");
         List<?> result = xpath.selectNodes(doc);
         assertEquals(3, result.size());
-        assertTrue(((org.w3c.dom.Node) result.get(0)).getNodeType() == Node.ATTRIBUTE_NODE);
-        assertTrue(((org.w3c.dom.Node) result.get(1)).getNodeType() == Pattern.NAMESPACE_NODE);
+        assertEquals(Node.ATTRIBUTE_NODE, ((Node) result.get(0)).getNodeType());
```

Comment:
should these two be assertSame? I suppose it depends whether Node.ATTRIBUTE_NODE is primitive or not

Reply:
If you want to use `assertSame`, Java has to auto-box this to no primitive `Short` and compare them.

Reply:
if both are primitives, this is fine. Just needed to check they weren't Java enums or something weird